web: revised package.json

kensternberg-authentik opened the pull request:
## Details

Revise the package.json file to separate the dependencies and devDependencies. 

## Changes

That's about it.  All code dependencies are in the `dependencies` block, and all tooling is in the `devDependencies` block.

## Checklist

-   [X] Local tests pass (`ak test authentik/`)
-   [X] The code has been formatted (`make lint-fix`)

If an API change has been made

-   An API change was not made.

If changes to the frontend have been made

-   This is a minor nitpick fix that does not affect the actual code, just the code's infrastructure quality.

-   The documentation was not changed.

kensternberg-authentik committed:
FOUND: Revise the package.json file

Reviewed the package.json file and separated out the dependencies
from the devDependencies.  Normally, this doesn't make a difference,
but when deployed as libraries the devDependencies don't normally
get considered by the deploying aplication.

kensternberg-authentik committed:
Merge branch 'main' into BUG-5171-revised-package-json

* main: (28 commits)
  root: migrate from os.path to Pathlib (#5594)
  events: don't include task uid in task metric (#5595)
  providers/scim: fix group patch schema (#5596)
  website: add email subscription form to footer (#5597)
  website/docs: added info about how to upgrade (#5589)
  website/docs: update 2023.5 release notes (#5526)
  web: bump pyright from 1.1.307 to 1.1.308 in /web (#5591)
  core: bump docker from 6.1.1 to 6.1.2 (#5592)
  core: bump django-otp from 1.1.6 to 1.2.0 (#5587)
  root: Test codeowners (#5586)
  stages/identification: auto-redirect to source when no user fields are selected (#5583)
  root: switch sentry dsn to our relay (#5494)
  blueprints: further copy-edits (#5559)
  sources/oauth: re-fix reddit source (#5582)
  web: bump API Client version (#5581)
  core: compile backend translations (#5580)
  core: compile backend translations (#5579)
  Translations for web/src/locales/en.po in zh_CN (#5574)
  Translations for web/src/locales/en.po in zh-Hans (#5575)
  Translations for locale/en/LC_MESSAGES/django.po in zh-Hans (#5576)
  ...

kensternberg-authentik committed:
Merge branch 'main' into BUG-5171-revised-package-json

* main:
  events: sanitize enums (#5610)
  blueprints: add meta models to schema (#5611)
  providers/scim: improve SCIM error messages (#5600)
  website/integrations: harbor requires a signing key for RS256 (#5601)
  web: bump @rollup/plugin-typescript from 11.1.0 to 11.1.1 in /web (#5605)
  web: bump @rollup/plugin-commonjs from 24.1.0 to 25.0.0 in /web (#5604)
  web: bump @codemirror/lang-javascript from 6.1.7 to 6.1.8 in /web (#5606)
  core: bump dacite from 1.8.0 to 1.8.1 (#5607)
  core: bump ruff from 0.0.265 to 0.0.267 (#5608)

kensternberg-authentik committed:
FIX Undo overrides and work-arounds for bad upstream postgres build.

kensternberg-authentik commented on web/package.json:
This is probably best viewed in its whole: **⋯ -> View file**

kensternberg-authentik committed:
5171: Fixed README to comply with Prettier rules.

I'm pretty impressed that that worked.  Good on Jens for having that
in the prettier rules.

BeryJu reviewed (commented):
The migration file should not be required (the choices aren't actually in the database so when there's a change to choices we usually just update the previous migration with that AlterField call, which was done on main)

Also the PR title should match the naming guide of https://goauthentik.io/developer-docs/#styleguides, so `web: revised package.json`

BeryJu commented on authentik/providers/oauth2/migrations/0017_alter_oauth2provider_sub_mode.py:
The file gets created by running `./manage.py makemigrations`, but you can just remove the file

kensternberg-authentik committed:
web: revised package.json

Removed the migration and web/README.md file.  The former should not have
been included; the latter is currently unprofessional in tone.

BeryJu committed:
fix docker build failing as we set NODE_ENV but need dev dependencies to build

Signed-off-by: Jens Langhammer <jens@goauthentik.io>
